### PR TITLE
Improved documentation for Raster-like elements

### PR DESCRIPTION
**Pull request by philippjfr**
Various improvements for the docstrings and documentation of gridded data and raster-like elements.

- [x] Closes https://github.com/ioam/holoviews/issues/2998
- [x] Closes https://github.com/ioam/holoviews/issues/1766

### CODE REVIEW

File: examples/reference/elements/bokeh/HSV.ipynb [hunk 1]
```diff
@@ -28,9 +28,15 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "HoloViews makes it trivial to  work in any color space that can be converted to ``RGB`` by making a simple subclass of ``RGB`` as appropriate. For instance, we also provide the  HSV (hue, saturation, value) color space, which is useful for plotting cyclic data (as the Hue) along with two additional dimensions (controlling the saturation and value of the color, respectively).\n",
+    "HSV represents a regularly spaced 2D grid of an underlying continuous space of HSV (hue, saturation and value) color space values. The definition of the grid closely matches the semantic of an ``Image`` or ``RGB`` element and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds, but it may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an HSV element therefore include:\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/HSV.ipynb]:
```suggestion
    "HSV represents a regularly spaced 2D grid of an underlying continuous space of HSV (hue, saturation and value) color space values. The definition of the grid closely matches the semantics of an ``Image`` or ``RGB`` element. In the simplest case the grid may be specified as an ``NxMx3`` or ``NxMx4`` array of values along with a bounds.  An HSV element may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an HSV element therefore include:\n",
```

File: examples/reference/elements/bokeh/HSV.ipynb [hunk 2]
```diff
@@ -28,9 +28,15 @@
...
     "\n",
-    "Like other raster based Element types ``HSV`` accepts gridded data, which may be supplied as a simple array ``NxMx3`` ndarray representing hue, saturation and value channels along with bounds or explicit array coordinates. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to see the accepted data formats."
+    "    HSV((X, Y, H, S, V))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``H``/``S``/``V`` are 2D array of shape ``NxM``, or equivalently:\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/HSV.ipynb]:
```suggestion
    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``H``/``S``/``V`` are 2D arrays of shape ``NxM``, or equivalently:\n",
```

File: examples/reference/elements/bokeh/HSV.ipynb [hunk 3]
```diff
@@ -28,9 +28,15 @@
...
+    "\n",
+    "    HSV(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where ``Z`` is a 3D array of stacked ``H``/``S``/``V`` arrays with shape ``NxMx3/4`` and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats."
```
Comment (jbednar) [inline examples/reference/elements/bokeh/HSV.ipynb]:
```suggestion
    "where ``Z`` is a 3D array of stacked ``H``/``S``/``V`` arrays with shape ``NxMx3/4`` and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats."
```

File: examples/reference/elements/bokeh/Image.ipynb [hunk 4]
```diff
@@ -28,7 +28,15 @@
...
+    "Image represents a regularly sampled 2D grid of an underlying [continuous space](Continuous_Coordinates.ipynb) of intensity values, which will be colormapped on plotting. The grid of intensity values may be specified as a ``NxM`` sized array of values along with a bounds, but it may also be defined through explicit and regularly spaced x/y-coordinate arrays of shape ``M`` and ``N`` respectively. The two most basic supported constructors of an Image therefore include:\n",
+    "\n",
+    "    Image((X, Y, Z))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``Z`` is a 2D array of shape ``NxM``, or equivalently:\n",
+    "\n",
+    "    Image(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where ``Z`` is a 2D array of shape ``NxM`` defining the intensity values and the bounds define the (left, bottom, top, right) edges of four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats."
```
Comment (jbednar) [inline examples/reference/elements/bokeh/Image.ipynb]:
```suggestion
    "where ``Z`` is a 2D array of shape ``NxM`` defining the intensity values and the bounds define the (left, bottom, top, right) edges of four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats."
```

File: examples/reference/elements/bokeh/QuadMesh.ipynb [hunk 5]
```diff
@@ -28,7 +28,17 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The basic ``QuadMesh`` is a 2D grid of bins specified as x-/y-values specifying a regular sampling or edges, with arbitrary sampling and an associated 2D array containing the bin values. Here is a simple ``QuadMesh`` with logarithmic sampling along the 'x' dimensions:"
+    "A ``QuadMesh`` represents 2D rectangular grid expressed as x- and y-coordinates defined as 1D or 2D arrays. Unlike the Image type a QuadMesh may be regularly or irregularly spaced and contain either bin edges or bin centers. If bin edges are supplied the shape of the x/y-coordinate arrays should be one greater than the value array.\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/QuadMesh.ipynb]:
```suggestion
    "A ``QuadMesh`` represents a 2D rectangular grid expressed as x- and y-coordinates defined as 1D or 2D arrays. Unlike an ``Image``, a QuadMesh may be regularly or irregularly spaced and contain either bin edges or bin centers. If bin edges are supplied, the shape of the x/y-coordinate arrays should be one greater than the value array.\n",
```

File: examples/reference/elements/bokeh/QuadMesh.ipynb [hunk 6]
```diff
@@ -28,7 +28,17 @@
...
+    "\n",
+    "The default interface expects data to be specified in the form:\n",
+    "\n",
+    "    QuadMesh((X, Y, Z))\n",
+    "\n",
+    "where ``X`` and ``Y`` may be 1D or 2D arrays of the shape ``N(+1)`` and ``M(+1)`` respectively or ``N(+1)xM(+1)`` and the ``Z`` value array should be of  shape NxM. Other gridded formats such as xarray are also supported if installed.\n",
+    "\n",
+    "The grid orientation follows the standard matrix convention: An array ``Z`` with shape (nrows, ncolumns) is plotted with the column number as ``X`` and the row number as ``Y``. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats.\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/QuadMesh.ipynb]:
```suggestion
    "The grid orientation follows the standard matrix convention: An array ``Z`` with shape (nrows, ncolumns) is plotted with the column number as ``X`` and the row number as ``Y``. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats.\n",
```

File: examples/reference/elements/bokeh/RGB.ipynb [hunk 7]
```diff
@@ -28,7 +28,17 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The ``RGB`` element is a subclass of [``Image``](./Image.ipynb) that supports red, green, blue channels. One of the simplest ways of creating an ``RGB`` element is to load an image file (such as PNG) off disk, using the ``load_image`` classmethod:"
+    "``RGB`` represents a regularly spaced 2D grid of an underlying continuous space of RGB(A) (red, green, blue and alpha) color space values. The definition of the grid closely matches the semantics of an Image and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds, but it may also be defined through explicit and regularly space x/y-coordinate arrays. The two most basic supported constructors of an ``RGB`` element therefore include:\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/RGB.ipynb]:
```suggestion
    "``RGB`` represents a regularly spaced 2D grid of an underlying continuous space of RGB(A) (red, green, blue, and alpha) color space values. The definition of the grid closely matches the semantics of an Image and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds.  An RGB may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an ``RGB`` element therefore include:\n",
```

File: examples/reference/elements/bokeh/RGB.ipynb [hunk 8]
```diff
@@ -28,7 +28,17 @@
...
+    "\n",
+    "    RGB((X, Y, R, G, B))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``R``/``G``/``B`` are 2D array of shape ``NxM``, or equivalently:\n",
+    "\n",
+    "    RGB(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where Z is a 3D array of stacked R/G/B arrays with shape NxMx3/4 and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats.\n",
```
Comment (jbednar) [inline examples/reference/elements/bokeh/RGB.ipynb]:
```suggestion
    "where Z is a 3D array of stacked R/G/B arrays with shape NxMx3/4 and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats.\n",
```

File: examples/reference/elements/matplotlib/HSV.ipynb [hunk 9]
```diff
@@ -28,9 +28,15 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "HoloViews makes it trivial to  work in any color space that can be converted to ``RGB`` by making a simple subclass of ``RGB`` as appropriate. For instance, we also provide the  HSV (hue, saturation, value) color space, which is useful for plotting cyclic data (as the Hue) along with two additional dimensions (controlling the saturation and value of the color, respectively).\n",
+    "HSV represents a regularly spaced 2D grid of an underlying continuous space of HSV (hue, saturation and value) color space values. The definition of the grid closely matches the semantic of an ``Image`` or ``RGB`` element and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds, but it may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an HSV element therefore include:\n",
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/HSV.ipynb]:
```suggestion
    "HSV represents a regularly spaced 2D grid of an underlying continuous space of HSV (hue, saturation and value) color space values. The definition of the grid closely matches the semantics of an ``Image`` or ``RGB`` element and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds. An HSV element may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an HSV element therefore include:\n",
```

File: examples/reference/elements/matplotlib/HSV.ipynb [hunk 10]
```diff
@@ -28,9 +28,15 @@
...
     "\n",
-    "Like other raster based Element types ``HSV`` accepts gridded data, which may be supplied as a simple array ``NxMx3`` ndarray representing hue, saturation and value channels along with bounds or explicit array coordinates. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to see the accepted data formats."
+    "    HSV((X, Y, H, S, V))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``H``/``S``/``V`` are 2D array of shape ``NxM``, or equivalently:\n",
+    "\n",
+    "    HSV(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where ``Z`` is a 3D array of stacked ``H``/``S``/``V`` arrays with shape ``NxMx3/4`` and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats."
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/HSV.ipynb]:
```suggestion
    "where ``Z`` is a 3D array of stacked ``H``/``S``/``V`` arrays with shape ``NxMx3/4`` and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats."
```

File: examples/reference/elements/matplotlib/Image.ipynb [hunk 11]
```diff
@@ -28,7 +28,15 @@
...
+    "Image represents a regularly sampled 2D grid of an underlying [continuous space](Continuous_Coordinates.ipynb) of intensity values, which will be colormapped on plotting. The grid of intensity values may be specified as a ``NxM`` sized array of values along with a bounds, but it may also be defined through explicit and regularly spaced x/y-coordinate arrays of shape ``M`` and ``N`` respectively. The two most basic supported constructors of an Image therefore include:\n",
+    "\n",
+    "    Image((X, Y, Z))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``Z`` is a 2D array of shape ``NxM``, or equivalently:\n",
+    "\n",
+    "    Image(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where ``Z`` is a 2D array of shape ``NxM`` defining the intensity values and the bounds define the (left, bottom, top, right) edges of four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats."
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/Image.ipynb]:
```suggestion
    "where ``Z`` is a 2D array of shape ``NxM`` defining the intensity values and the bounds define the (left, bottom, top, right) edges of four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats."
```

File: examples/reference/elements/matplotlib/QuadMesh.ipynb [hunk 12]
```diff
@@ -28,7 +28,17 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The basic ``QuadMesh`` is a 2D grid of bins specified as x-/y-values specifying a regular sampling or edges, with arbitrary sampling and an associated 2D array containing the bin values. Here is a simple ``QuadMesh`` with logarithmic sampling along the 'x' dimensions:"
+    "A ``QuadMesh`` represents 2D rectangular grid expressed as x- and y-coordinates defined as 1D or 2D arrays. Unlike the Image type a QuadMesh may be regularly or irregularly spaced and contain either bin edges or bin centers. If bin edges are supplied the shape of the x/y-coordinate arrays should be one greater than the value array.\n",
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/QuadMesh.ipynb]:
```suggestion
    "A ``QuadMesh`` represents 2D rectangular grid expressed as x- and y-coordinates defined as 1D or 2D arrays. Unlike the Image type, a QuadMesh may be regularly or irregularly spaced and contain either bin edges or bin centers. If bin edges are supplied, the shape of the x/y-coordinate arrays should be one greater than the value array shape.\n",
```

File: examples/reference/elements/matplotlib/QuadMesh.ipynb [hunk 13]
```diff
@@ -28,7 +28,17 @@
...
+    "\n",
+    "The default interface expects data to be specified in the form:\n",
+    "\n",
+    "    QuadMesh((X, Y, Z))\n",
+    "\n",
+    "where ``X`` and ``Y`` may be 1D or 2D arrays of the shape ``N(+1)`` and ``M(+1)`` respectively or ``N(+1)xM(+1)`` and the ``Z`` value array should be of  shape NxM. Other gridded formats such as xarray are also supported if installed.\n",
+    "\n",
+    "The grid orientation follows the standard matrix convention: An array ``Z`` with shape (nrows, ncolumns) is plotted with the column number as ``X`` and the row number as ``Y``. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats.\n",
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/QuadMesh.ipynb]:
```suggestion
    "The grid orientation follows the standard matrix convention: An array ``Z`` with shape (nrows, ncolumns) is plotted with the column number as ``X`` and the row number as ``Y``. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats.\n",
```

File: examples/reference/elements/matplotlib/RGB.ipynb [hunk 14]
```diff
@@ -28,7 +28,17 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The ``RGB`` element is a subclass of [``Image``](./Image.ipynb) that supports red, green, blue channels. One of the simplest ways of creating an ``RGB`` element is to load an image file (such as PNG) off disk, using the ``load_image`` classmethod:"
+    "``RGB`` represents a regularly spaced 2D grid of an underlying continuous space of RGB(A) (red, green, blue and alpha) color space values. The definition of the grid closely matches the semantics of an Image and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds, but it may also be defined through explicit and regularly space x/y-coordinate arrays. The two most basic supported constructors of an ``RGB`` element therefore include:\n",
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/RGB.ipynb]:
```suggestion
    "``RGB`` represents a regularly spaced 2D grid of an underlying continuous space of RGB(A) (red, green, blue and alpha) color space values. The definition of the grid closely matches the semantics of an Image and in the simplest case the grid may be specified as a ``NxMx3`` or ``NxMx4`` array of values along with a bounds, but it may also be defined through explicit and regularly spaced x/y-coordinate arrays. The two most basic supported constructors of an ``RGB`` element therefore include:\n",
```

File: examples/reference/elements/matplotlib/RGB.ipynb [hunk 15]
```diff
@@ -28,7 +28,17 @@
...
+    "\n",
+    "    RGB((X, Y, R, G, B))\n",
+    "\n",
+    "where ``X`` is a 1D array of shape ``M``, ``Y`` is a 1D array of shape ``N`` and ``R``/``G``/``B`` are 2D array of shape ``NxM``, or equivalently:\n",
+    "\n",
+    "    RGB(Z, bounds=(x0, y0, x1, y1))\n",
+    "\n",
+    "where Z is a 3D array of stacked R/G/B arrays with shape NxMx3/4 and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide to all the other accepted data formats.\n",
```
Comment (jbednar) [inline examples/reference/elements/matplotlib/RGB.ipynb]:
```suggestion
    "where Z is a 3D array of stacked R/G/B arrays with shape NxMx3/4 and the bounds define the (left, bottom, top, right) edges of the four corners of the grid. Other gridded formats which support declaring of explicit x/y-coordinate arrays such as xarray are also supported. See the [Gridded Datasets](../../../user_guide/08-Gridded_Datasets.ipynb) user guide for all the other accepted data formats.\n",
```

File: examples/user_guide/08-Gridded_Datasets.ipynb [hunk 16]
```diff
@@ -30,6 +31,17 @@
     "* they have coordinate or key dimensions, which describe the sampling of each dimension in the value arrays\n",
     "* they have value dimensions which describe the quantity of the multi-dimensional value arrays\n",
     "\n",
+    "There are many different types of gridded datasets defined by their coordinates, however in holoviews gridded datasets mostly refers to two types of grids usually known as regular and irregular coordinates or rectilinear or curvilinear. Rectilinear grids are defined by 1D coordinate arrays while rectilinear grid coordinates have the same dimensionality as the underlying value arrays. Among rectilinear grids there are two different subtypes, uniform and irregularly spaced grid. Only some types of elements support irregularly spaced rectilinear and curvilinear grids, specifically a QuadMesh is usually used to represent these.\n",
```
Comment (jbednar) [inline examples/user_guide/08-Gridded_Datasets.ipynb]:
```suggestion
    "There are many different types of gridded datasets, which each approximate or measure a surface or space at discretely specified coordinates.  In HoloViews, gridded datasets are typically one of three possible types: Regular rectilinear grids, irregular rectilinear grids, and curvilinear grids.  Regular rectilinear grids can be defined by 1D coordinate arrays specifying the spacing along each dimension, while the other types require grid coordinates with the same dimensionality as the underlying value arrays, specifying the full n-dimensional coordinates of the corresponding array value. HoloViews provides many different elements supporting regularly spaced rectilinear grids, but currently only QuadMesh supports irregularly spaced rectilinear and curvilinear grids.\n",
```

(Please edit as needed; I'm not certain this captures the actual behavior, but hopefully you get the idea. In the original I was getting very confused about whether there were two types of grids each with two names, four types, three types, etc. Here I'm assuming there are three types, but if irregular curvilinear is also an option distinct from just curvilinear, then please add that as a fourth type.  )

File: examples/user_guide/08-Gridded_Datasets.ipynb [hunk 17]
```diff
@@ -293,6 +305,63 @@
     "heatmap + heatmap.table()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Non-uniform rectilinear grids\n",
+    "\n",
+    "As discussed above, there are two main types of grids handled by HoloViews, so far we have mainly dealt with uniform, rectilinear grids. However using the ``QuadMesh`` element non-uniform rectilinear grids and curvilinear grids can also be visualized.\n",
```
Comment (jbednar) [inline examples/user_guide/08-Gridded_Datasets.ipynb]:
```suggestion
    "As discussed above, there are two main types of grids handled by HoloViews.  So far, we have mainly dealt with uniform, rectilinear grids, but we can use the ``QuadMesh`` element to work with non-uniform rectilinear grids and curvilinear grids.\n",
```

File: examples/user_guide/08-Gridded_Datasets.ipynb [hunk 18]
```diff
@@ -352,6 +421,58 @@
    "source": [
     "Here, we have explicitly specified the default behaviour `dynamic=False`, which returns a HoloMap. Note, that this approach immediately converts all available data to images, which will take up a lot of RAM for large datasets. For these situations, use `dynamic=True` to generate a [DynamicMap](./06-Live_Data.ipynb) instead. Additionally, [xarray features dask support](http://xarray.pydata.org/en/stable/dask.html), which is helpful when dealing with large amounts of data.\n",
     "\n",
+    "It is also possible to render curvilinear grids with xarray, to demonstrate this we will load one such example. The dataset below defines a curvilinear grid of air temperatures varying over time. The curvilinear grid can be identified by the fact that the ``xc`` and ``yc`` coordinates are defined as two-dimensional arrays:"
```
Comment (jbednar) [inline examples/user_guide/08-Gridded_Datasets.ipynb]:
```suggestion
    "It is also possible to render curvilinear grids with xarray, and here we will load one such example. The dataset below defines a curvilinear grid of air temperatures varying over time. The curvilinear grid can be identified by the fact that the ``xc`` and ``yc`` coordinates are defined as two-dimensional arrays:"
```

File: examples/user_guide/08-Gridded_Datasets.ipynb [hunk 19]
```diff
@@ -352,6 +421,58 @@
...
+   "source": [
+    "rasm = rasm.isel(time=0, x=slice(0, 200)).assign_coords(x=np.arange(200), y=np.arange(205))\n",
+    "rasm.coords"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have both logical, rectilinear and curvilinear coordinates we can visualize the difference between the two by explicitly defining which set of coordinates to use:"
```
Comment (jbednar) [inline examples/user_guide/08-Gridded_Datasets.ipynb]: I'm not sure what that means -- maybe rephrase as:

```suggestion
    "Now that we have defined both rectilinear and curvilinear coordinates we can visualize the difference between the two by explicitly defining which set of coordinates to use:"
```

File: holoviews/element/raster.py [hunk 20]
```diff
@@ -716,16 +775,25 @@ def rgb(self):
...
-    As a secondary but less supported mode QuadMesh can contain
-    a mesh of quadrilateral coordinates that is not laid out in
-    a grid. The data should then be supplied as three separate
-    2D arrays for the x-/y-coordinates and grid values.
+    A QuadMesh represents 2D rectangular grid expressed as x- and
+    y-coordinates defined as 1D or 2D arrays. Unlike the Image type
+    a QuadMesh may be regularly or irregularly spaced and contain
+    either bin edges or bin centers. If bin edges are supplied the
+    shape of the x/y-coordinate arrays should be one greater than the
+    value array.
```
Comment (jbednar) [inline holoviews/element/raster.py]:
```suggestion
    shape of the value array.
```

File: holoviews/element/raster.py [hunk 21]
```diff
@@ -810,12 +878,21 @@ def trimesh(self):
 
 class HeatMap(Dataset, Element2D):
     """
-    HeatMap is an atomic Element used to visualize two dimensional
-    parameter spaces. It supports sparse or non-linear spaces, dynamically
-    upsampling them to a dense representation, which can be visualized.
+    HeatMap represents a 2D grid of categorical coordinates which can
+    be computed from a sparse tabular representation. A HeatMap does
+    not automatically aggregate the supplied values, if the data
```
Comment (jbednar) [inline holoviews/element/raster.py]:
```suggestion
    not automatically aggregate the supplied values, so if the data
```

File: holoviews/element/raster.py [hunk 22]
```diff
@@ -810,12 +878,21 @@ def trimesh(self):
...
+    contains multiple entries for the same coordinate on the 2D grid
+    it should be aggregated using the aggregate method before display.
+
+    The HeatMap constructor will support any tabular or gridded data
+    format with 2 coordinate and at least one value dimensions. A
```
Comment (jbednar) [inline holoviews/element/raster.py]:
```suggestion
    format with 2 coordinates and at least one value dimension. A
```

(Maybe?)

File: examples/user_guide/08-Gridded_Datasets.ipynb (hunk 16 above)
Comment (philippjfr) [reply examples/user_guide/08-Gridded_Datasets.ipynb]: No you got it right. Initially I'd written it as two types, where uniform rectilinear are simply a subset of rectilinear grids in general.

**Comment by philippjfr** [comment]
Perfect, all the edits looked good. I'll go ahead and merge this once tests pass.

**Comment by jbednar** [comment]
Github  suggestions sure end up with a lot of commits!  Better remember to squash merge...

**Comment by philippjfr** [comment]
>Github suggestions sure end up with a lot of commits!

Yes, would be nice to have a UI to select suggestions and then apply them all at once. Either way still more convenient than manually applying these changes myself and we always squash merge in HoloViews.

**Comment by philippjfr** [comment]
Just noting that this PR also removed the deprecated ``HeatMap.raster`` property and ``GridImage``, which should be noted in the release notes.
